### PR TITLE
Support for Zipkin: CombineTraces

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
Uses span.Kind to differentiate between zipkin client/server spans in combine.

**Which issue(s) this PR fixes**:
Related to #682 and #687 

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/util/trace.go [hunk 1]
```diff
@@ -68,12 +69,13 @@ func CombineTraceProtos(traceA, traceB *tempopb.Trace) (*tempopb.Trace, int, int
 	spanCountTotal := 0
 
 	h := fnv.New32()
+	buffer := make([]byte, 4)
```
Comment (annanay25) [inline pkg/util/trace.go:72]: Just mentioning for reference (was wondering why we need 4bytes), the span kind field is a int32 
https://github.com/open-telemetry/opentelemetry-collector/blob/51661b8400d8a90e1bded85e2770fd56865761be/internal/data/protogen/trace/v1/trace.pb.go#L34

**Review by annanay25 — APPROVED** [review]
lgtm, nice!

File: pkg/util/trace.go (hunk 1 above)
Comment (joe-elliott) [reply pkg/util/trace.go:72]: 32 / 8 = 4?

Comment (annanay25) [reply pkg/util/trace.go:72]: lol i meant i didn't think we need more than 1 byte

**Pull request merged**